no-fallthrough failing on continue statement

Take a look at the following code

``` javascript
var i;
for (i = 0; i < 1; i += 1) {
    switch (i) {
    case 1:
        continue;
    default:
        break;
    }
}
```

When I run eslint with the `no-fallthrough` rule turned on, I get the following error

``` bash
test.js
  4:4  error  Expected a "break" statement before "default"  no-fallthrough
```

To fix the issue, I added a `break;` statement as recommended

``` javascript
var i;
for (i = 0; i < 1; i += 1) {
    switch (i) {
    case 1:
        continue;
        break;
    default:
        break;
    }
}
```

This time, I get another error:

``` bash
test.js
  6:8  error  Found unexpected statement after a continue  no-unreachable
```

Although I can work around this by using a `//falls through` comment instead of the `break;`, the code is technically never falling through and the comment will be misleading in that case.

--- a/lib/rules/no-fallthrough.js
+++ b/lib/rules/no-fallthrough.js
@@ -55,7 +55,7 @@ module.exports = function(context) {
             // try to verify using statements - go backwards as a fast path for the search
             if (consequent.length) {
                 for (i = consequent.length - 1; i >= 0; i--) {
-                    if (/(?:Break|Return|Throw)Statement/.test(consequent[i].type)) {
+                    if (/(?:Break|Continue|Return|Throw)Statement/.test(consequent[i].type)) {
                         switchData.lastCaseClosed = true;
                         break;
                     }

--- a/tests/lib/rules/no-fallthrough.js
+++ b/tests/lib/rules/no-fallthrough.js
@@ -22,6 +22,7 @@ eslintTester.addRuleTest("lib/rules/no-fallthrough", {
         "switch(foo) { case 0: a(); /* falls through */ case 1: b(); }",
         "function foo() { switch(foo) { case 0: a(); return; case 1: b(); }; }",
         "switch(foo) { case 0: a(); throw 'foo'; case 1: b(); }",
+        "while (a) { switch(foo) { case 0: a(); continue; case 1: b(); } }",
         "switch(foo) { case 0: a(); break; case 1: b(); }",
         "switch(foo) { case 0: case 1: a(); break; case 2: b(); }",
         "switch(foo) { case 0: case 1: break; case 2: b(); }",

